microcluster/app.go

```diff
@@ -480,3 +480,23 @@ func (m *MicroCluster) SQL(ctx context.Context, query string) (string, *internal
 
 	return "", batch, err
 }
+
+// APIExtensions returns the list of API extensions registered with the daemon.
+func (m *MicroCluster) APIExtensions(ctx context.Context) ([]string, error) {
+	c, err := m.LocalClient()
+	if err != nil {
+		return nil, err
+	}
+
+	clusterMembers, err := c.GetClusterMembers(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(clusterMembers) == 0 {
+		return nil, fmt.Errorf("No cluster members found when fetching API extensions")
+	}
+
+	// Each cluster member will have the same extensions, so we only need to check one.
+	return clusterMembers[0].Extensions, nil
```

Comment:
If the members have different API extensions I guess this will report only the extensions of a single member? Would it be worth returning a map instead that contains the list of extensions for every member?

Reply:
This would presumably happen during an update but I am not sure if during this time you could actually query the endpoint and get back a result?

Reply:
yes, let's do that it seems much more secure (though it looks like a very rare scenario). Then I'll add a validation logic on the MicroCloud side of things to ensure that all the members have the same extensions. :)

Reply:
@roosterfish updated